### ISSUE TITLE
Example tests failing

### ISSUE THREAD

**Issue by timblackwell** [snikch/goodman#14]
When running `make -C example dredd` the tests fail. 
Dredd 2.2.5 or using 1.6.1 from node_modules folder
Go 1.7
Node 7.3.0
NPM 3.10.10

Can someone re-run the Travis CI build/test to confirm that is it just my issue?

```
goodman ➤ make -C example dredd                                                                                                                                                                                                                                                 git:master* ../node_modules/.bin/dredd api.apib http://localhost:8080 --language go --server ./server --hookfiles hooks/hooks
info: Starting server with command: ./server
info: Waiting 3 seconds for server command to start...
info: Beginning Dredd testing...
info: Found Hookfiles: hooks/hooks
info: Spawning `go` hooks handler
info: Hook handler stdout: Sending to channel

Completed

info: Hook handler stderr: 2016/12/23 10:32:10 Starting hooks server in go routine

info: Hook handler stderr: 2016/12/23 10:32:10 method After reply type not a pointer: hooks.Callback
2016/12/23 10:32:10 method AfterAll has wrong number of ins: 2
2016/12/23 10:32:10 method AfterEach has wrong number of ins: 2
2016/12/23 10:32:10 method Before reply type not a pointer: hooks.Callback
2016/12/23 10:32:10 method BeforeAll has wrong number of ins: 2

info: Hook handler stderr: 2016/12/23 10:32:10 method BeforeEach has wrong number of ins: 2
2016/12/23 10:32:10 method BeforeEachValidation has wrong number of ins: 2
2016/12/23 10:32:10 method BeforeValidation reply type not a pointer: hooks.Callback

info: Hook handler stdout: Starting

info: Hook handler stdout: Accepting connection

info: Connected to the hook handler, waiting 0.1s to start testing.
info: Hook handler stderr: 2016/12/23 10:32:11 table users already exists

info: Hook handler stdout: Hooks client failed with exit status 1

info: Hook handler stderr: panic: RPC client threw error unexpected EOF

goroutine 1 [running]:
panic(0x2d6e20, 0xc420075f90
info: Hook handler stderr: )
	/usr/local/Cellar/go/1.7/libexec/src/runtime/panic.go:500 +0x1a1
github.com/snikch/goodman.(*Run).RunBeforeAll(0xc4200ef500, 0xc420070460)
	/Users/timothy.blackwell/go/src/github.com/snikch/goodman/runner.go:32 +0x1bc
github.com/snikch/goodman.(*Server).RunBeforeAll(0xc42004bee8, 0xc420070460)
	/Users/timothy.blackwell/go/src/github.com/snikch/goodman/server.go:135 +0x5e
github.com/snikch/goodman.(*Server).ProcessMessage(0xc42004bee8, 0xc420070420, 0x500, 0x2c6e20)
	/Users/timothy.blackwell/go/src/github.com/snikch/goodman/server.go:98 +0x5a2
github.com/snikch/goodman.(*Server).Run(0xc42004bee8, 0x0, 0x0)
	/Users/timothy.blackwell/go/src/github.com/snikch/goodman/server.go:70 +0x484
main.main()
	/Users/timothy.blackwell/go/src/github.com/snikch/goodman/cmd/goodman/main.go:60 +0x6d7

error: Hook handler '/Users/timothy.blackwell/go/bin/goodman' exited with status: 2
warn: Hook handling timed out.
error: Hook handler '/Users/timothy.blackwell/go/bin/goodman' exited with status: 2
info: Sending SIGTERM to the backend server
info: Backend server was killed
make: *** [dredd] Error 1 
```

**Comment by ddelnano** [comment]
Go 1.7 is not tested on Travis at the moment.  I added it to the ci build just now here #15.  We will see if that passes.

**Comment by ddelnano** [comment]
@timblackwell Travis build passes with Go 1.7 so that can't be the issue.  The Travis build runs the example you are trying to run if you weren't aware.  Looking at your output I see something that looks problematic.  

```
info: Hook handler stderr: 2016/12/23 10:32:11 table users already exists
```

This looks like a previous run of the hooks created the database but crashed before the after all hooks were run.  Can you remove the `foo.db` and re run the example using the version of dredd from the node_modules?

**Comment by ddelnano** [comment]
@timblackwell also just to make you aware doesn't look like node v7 is officially supported by dredd.  See the travis.yml [here](https://github.com/apiaryio/dredd/blob/master/.travis.yml#L8).

I would say remove that db file, switch to a node version that they test against, use the dredd version in the node_modules folder and let me know how that goes.

**Comment by timblackwell** [comment]
Hi @ddelnano, thanks for the pointers. The current issue was the `foo.db` but the route cause was the node version, just as you suggested.  Testing on node v6.9.2 works, strange that the python hooks work with Dredd on node 7.3.0 though...

**Comment by ddelnano** [comment]
Cool no problem.  I wonder if you were using all the same hook lifecycle events (before each, after each, etc) as the python ones.

**Comment by timblackwell** [comment]
I only need the before event, will drop down to node 6.9.2 and try again in
my project.

On 24 Dec 2016 8:09 pm, "Dom Del Nano" <notifications@github.com> wrote:

Cool no problem. I wonder if you were using all the same hook lifecycle
events (before each, after each, etc) as the python ones.

—
You are receiving this because you modified the open/close state.

Reply to this email directly, view it on GitHub
<https://github.com/snikch/goodman/issues/14#issuecomment-269098514>, or mute
the thread

**Comment by radimk** [comment]
Just a clarification since I hit the same problem: I use Debian and installed nodejs package version 7 which lead me to the problem described above. I uninstalled it and installed node 6 (more precisely v6.9.5) using nvm. `npm uninstall dredd && npm install` didn't help. I had to delete `foo.db` file in examples folder to let the test pass.

**Comment by ddelnano** [comment]
@radimk so the node 7 is all on Dredd itself since it doesn't support it as I've said above.  As for the database file that sticks around I created #21 to address the problem.  I just threw that together quickly but should merge it later today after I do a test to verify it does fix this issue.

**Comment by ddelnano** [comment]
@radimk @timblackwell a known issue with node 7 that caused issues with the Go hooks was merged [here](https://github.com/apiaryio/dredd/pull/750).  Released in version 3.2.1. Just thought I would give you an update.  They are going to add node 7 to the Travis build so it should be working fully soon.
